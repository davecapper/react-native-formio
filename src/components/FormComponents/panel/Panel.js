import React from 'react';
// import {FormioComponentsList} from '../../';
import {View} from 'react-native';
import {Card} from '@rneui/themed';
import PropTypes from 'prop-types';
import {StyleSheet} from 'react-native';
import styles from './styles';

const Panel = (props) => {
  const title = (props.component.title && !props.component.hideLabel ? props.component.title : undefined);
  const titleStyle = {...StyleSheet.flatten(styles.title), color: props.colors.secondaryTextColor};

    return (
      <Card containerStyle={styles.panel}>
        <Card.Title style={titleStyle}>{title}</Card.Title>
        <View style={styles.componentsWrapper}>
          {/* <FormioComponentsList
            {...props}
            components={props.component.components}
          ></FormioComponentsList> */}
        </View>
      </Card>
    );
};

Panel.propTypes = {
  component: PropTypes.object,
  theme: PropTypes.object,
  colors: PropTypes.object
};

export default Panel;
