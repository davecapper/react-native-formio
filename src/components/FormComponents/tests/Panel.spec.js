import React from 'react';
import renderer from 'react-test-renderer';
import Panel from '../panel/Panel';
import colors from '../../../defaultTheme/colors';
import theme from '../../../defaultTheme';

import * as components from '../../../../test/forms/componentSpec';

describe('Panel', () => {
  describe(' Panel component', () => {
    var component= {
      'input': false,
      'title': 'testpanel',
      'theme': 'warning',
      'components': [
        components.textfield,
        components.password,
        components.phoneNumber
      ],
      'type': 'panel',
      'conditional': {
        'show': '',
        'when': null,
        'eq': ''
      }
    };
    const attachToForm = jest.fn();

    it.only('Renders a basic panel component', () => {
      const element = renderer.create(<Panel
        component={component}
        attachToForm={attachToForm}
        colors={colors}
        theme={theme}
      />);

      expect(element).toMatchSnapshot();
      // expect(element.find('.panel').length).to.equal(1);
    });

    it('Test the theme of panel component', function(done) {
      const element = renderer.create(<Panel
        component={component}
        attachToForm={attachToForm}
        />).find('.panel');
      expect(element.nodes[0].props.className).to.equal('panel panel-' + component.theme + ' ');
      done();
    });

    it('Test the label of panel component', function(done) {
      const element = renderer.create(<Panel
        component={component}
        attachToForm={attachToForm}
        />).find('.panel');
      expect(element.nodes[0].props.children[0].props.className).to.equal('panel-heading');
      expect(element.nodes[0].props.children[0].props.children.props.className).to.equal('panel-title');
      done();
    });

    it('Check the nested components of Panel', function(done) {
      const element = renderer.create(<Panel
        component={component}
        attachToForm={attachToForm}
        />).find('.panel-body');

      //To test type of nested components of panel
      for (var i= 0; i<component.components.length; i++) {
        expect(element.nodes[0].props.children.props.component.components[i].type).to.equal(component.components[i].type);
      }
      done();
    });
  });
});
