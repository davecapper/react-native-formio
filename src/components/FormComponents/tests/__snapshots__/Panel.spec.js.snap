// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel  Panel component Renders a basic panel component 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "margin": 15,
      "marginBottom": 0,
      "padding": 0,
      "paddingBottom": 20,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "transparent",
      }
    }
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "alignSelf": "flex-start",
          "color": "rgba(0, 0, 0, 0.54)",
          "fontSize": 18,
          "fontWeight": "bold",
          "margin": 10,
          "marginBottom": 15,
          "textAlign": "center",
        }
      }
      testID="cardTitle"
    >
      testpanel
    </Text>
    <View
      style={
        {
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
